content/guides/intermediate/shaders.md

```diff
@@ -0,0 +1,662 @@
+---
+title: "Shaders"
+authors: [Jasper]
+date: 2025-02-20
+---
+
+> [!CAUTION]
+> This guide is made for löve 12.0!
+
+With **shaders** we can create fun graphical effects for our games.
+
+There are multiple types of shaders available to be used in löve
+* The **Fragment**, AKA **Pixel** shader
+* The **Vertex** shader
+* The **Compute** shader
+
+A shader is a piece of code the GPU executes, like when drawing images or text.
+
+Shaders can be created using `love.graphics.newShader` and `love.graphics.newComputeShader`.
+Compute shaders won't be covered here.
+
+## Syntax
+
+löve shaders are written in GLSL, which is quite a bit different to lua.
+Some notable difference are that GLSL is statically typed, semicolon line breaks, curly brackets to define code blocks instead of `then / do`, `end`.       
+
+There are some limitations to take in to consideration when writing shader code, the main one is that, due to the way GPUs work the entire register usage needs to be known beforehand, meaning dynamic memory allocations like these:
+```glsl
+uniform int IntCount;
+uniform int[IntCount] UniformIntegers
+```
+aren't allowed.     
+Which leads to another limitation, recursive code is not allowed either, use a stack and a while loop instead.
+
+### Types
+
+Scalar types:
+* `float`: 32-bit floating point number, for storing fractional values
+* `int`: 32-bit integer value
+* `uint`: unsigned version of int, meaning it can't be negative, allowing for twice the range of values
+* `bool`: boolean value (`true` or `false`)
+
+Vector types:
+* `vecn`: float type vector
+* `ivecn`: integer type vector
+* `uvecn`: unsigned integer type vector
+* `bvecn`: boolean type vector
+
+Matrix types:
+* mat*n*x*m*: n columns, m rows (column major) 
+
+>[!IMPORTANT]
+> When doing math in shaders, make sure to use the correct literal number formatting        
+> `1 / 2 = 0` Values without type suffixes will be interpreted as integers      
+> `1.0 / 2.0 = 0.5` Floating point division
+
+Type suffixes
+* `float`: *x*.;*x*.0;*x*.f (1. ; 1.0 ; 1.f)
+* `int`: *x* ( 1 )
+* `uint` *x*u ( 1u )
+
+[This](https://learnwebgl.brown37.net/12_shader_language/glsl_mathematical_operations.html) page explains operators.
+
+## The Fragment shader
+This shader is executed for every pixel the effect covers.
+Let's make a simple fragment shader from a string to show it's functionality.
+
+---
+
+### Simple color
+
+```lua
+local shaderCode = [[
+vec4 effect(vec4 color, Image tex, vec2 texture_coords, vec2 screen_coords)
```

Comment:
nitpick: its not your fault at all but i wonder if it's better to split these blocks up just so we can get glsl highlighting? as this is the introduction to shaders it might make it easier to read for anyone unfamilar with glsl (but still note that you can load strings directly like this?)